templates/components/form/reservation.html.twig

```diff
@@ -0,0 +1,188 @@
+{#
+ # ---------------------------------------------------------------------
+ #
+ # GLPI - Gestionnaire Libre de Parc Informatique
+ #
+ # http://glpi-project.org
+ #
+ # @copyright 2015-2023 Teclib' and contributors.
+ # @copyright 2003-2014 by the INDEPNET Development Team.
+ # @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ #
+ # ---------------------------------------------------------------------
+ #
+ # LICENSE
+ #
+ # This file is part of GLPI.
+ #
+ # This program is free software: you can redistribute it and/or modify
+ # it under the terms of the GNU General Public License as published by
+ # the Free Software Foundation, either version 3 of the License, or
+ # (at your option) any later version.
+ #
+ # This program is distributed in the hope that it will be useful,
+ # but WITHOUT ANY WARRANTY; without even the implied warranty of
+ # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ # GNU General Public License for more details.
+ #
+ # You should have received a copy of the GNU General Public License
+ # along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ #
+ # ---------------------------------------------------------------------
+ #}
+
+{% extends "generic_show_form.html.twig" %}
+{% import 'components/form/fields_macros.html.twig' as fields %}
+
+{% block form_fields %}
+    {% if items|length > 0 %}
+        {% for item in items %}
+            {{ fields.htmlField(
+                '',
+                item.type_name,
+                _n('Item', 'Items', items|length)
+            ) }}
+            {{ fields.htmlField(
+                '',
+                item.comment,
+                __('Notes')
+            ) }}
+            {{ fields.hiddenField(
+                'items[' ~ item.id ~ ']',
+                item.id,
+                '',
+                {
+                    add_field_class: 'd-none'
+                }
+            ) }}
+        {% endfor %}
+    {% else %}
+        {{ fields.dropdownItemsFromItemtypes(
+            'items[]',
+            _n('Item', 'Items', 1),
+            {
+                items_id_name   : 'items[]',
+                itemtypes       : itemtypes,
+                entity_restrict : session('glpiactive_entity'),
+                checkright      : false,
+                ajax_page       : config('root_doc') ~ '/ajax/reservable_items.php'
+            }
+        ) }}
+    {% endif %}
+
+    {{ fields.dropdownField(
+        'User',
+        'users_id',
+        item.fields['users_id'],
+        __('By'),
+        {
+            entity      : item.fields['entities_id'],
+            right       : 'all'
+        }
+    ) }}
```

Comment:
1. On new item creation, value should fallback to current connected user (code was `$uid = (empty($ID) ? Session::getLoginUserID() : $resa->fields['users_id']);`).
2. Field should be readonly in user has no right to update the reservation (there was a condition on `!Session::haveRight("reservation", UPDATE) || !Session::haveAccessToEntity($entities_id)`).